python/mxnet/operator.py

```diff
@@ -522,6 +525,71 @@ def infer_type(self, in_type):
         return in_type, [in_type[0]]*len(self.list_outputs()), \
             [in_type[0]]*len(self.list_auxiliary_states())
 
+    def infer_storage_type(self, in_stype):
+        """infer_storage_type interface. Used to infer storage type of
+        inputs and outputs in the forward pass.
+
+        Parameters
+        ----------
+        in_stype : list of stypes, Valid stypes are default, row_sparse and
+            csr
+
+        Returns
+        -------
+        in_stype : list
+            list of argument stypes.
+        out_stype : list
+            list of output types calculated from in_stype,
+            in the same order as declared in list_outputs.
+        aux_type : Optional, list
+            list of aux types calculated from in_stype,
+            in the same order as declared in list_auxiliary_states.
+        """
+        return in_stype, [in_stype[0]]*len(self.list_outputs()), \
+            [in_stype[0]]*len(self.list_auxiliary_states())
+
+    def infer_storage_type_backward(self, ograd_stype, in_stype, out_stype, igrad_stype, aux_stype):
+        """infer_storage_type_backward interface. Used to infer storage
+        type of inputs and outputs in the backward pass.
+
+        If "undefined" is provided for any storage types it will fallback
+        to the existing storage type. If the existing storage type is
+        also "undefined" it will fallback to "default".
+        If returned lists are empty or less than the length of corresponding
+        input lists the missing values are populated from the input or fallback
+        to "default" if input storage types are "undefined".
+        If returned lists are greater than the length of corresponding input
+        input lists, it will throw an exception.
+
+        Parameters
+        ----------
+        ograd_stype : list
+            list of output gradient storage types.
+        in_stype : list
+            list of input storage types
+        out_stype : list
+            list of output storage types
+        igrad_stype : list
+            list of input gradient storage types.
+        aux_stype : list
+            list of auxiliary storage types
+
+        Returns
+        -------
+        ograd_stype : list
+            list of inferred output gradient storage types
+        in_stype : list
+            list of inferred input storage types
+        out_stype : list
+            list of inferred output storage types
+        igrad_stype : list
+            list of inferred input gradient storage types
+        aux_stype : list
+            list of inferred storage types for auxiliary states.
+        """
+        return list(ograd_stype), list(in_stype), list(out_stype), \
```

Comment:
This default implementation didn't do anything

Reply:
changed.